hoodie-common/src/main/java/com/uber/hoodie/common/table/view/HoodieTableFileSystemView.java

```diff
@@ -165,19 +178,47 @@ private void writeObject(java.io.ObjectOutputStream out)
     return Arrays.stream(statuses).filter(rtFilePredicate).map(HoodieLogFile::new);
   }
 
+  /**
+   * With concurrent compaction, it is possible to see partial/complete data-files due to inflight-compactions, Ignore
+   * those data-files
+   *
+   * @param dataFileOpt Data File
+   */
+  private boolean isDataFileNotDueToPendingCompaction(Optional<HoodieDataFile> dataFileOpt) {
+    if (dataFileOpt.isPresent()) {
+      HoodieDataFile dataFile = dataFileOpt.get();
+      return isDataFileNotDueToPendingCompaction(dataFile);
+    }
+    return false;
+  }
+
+  /**
+   * With concurrent compaction, it is possible to see partial/complete data-files due to inflight-compactions, Ignore
+   * those data-files
+   *
+   * @param dataFile Data File
+   */
+  private boolean isDataFileNotDueToPendingCompaction(HoodieDataFile dataFile) {
+    String compactionInstantTime = fileIdToPendingCompactionInstantTime.get(dataFile.getFileId());
+    if ((null != compactionInstantTime) && dataFile.getCommitTime().equals(compactionInstantTime)) {
+      return false;
+    }
+    return true;
+  }
+
   @Override
   public Stream<HoodieDataFile> getLatestDataFiles(final String partitionPath) {
     return getAllFileGroups(partitionPath)
         .map(fileGroup -> fileGroup.getLatestDataFile())
-        .filter(dataFileOpt -> dataFileOpt.isPresent())
+        .filter(this::isDataFileNotDueToPendingCompaction)
```

Comment:
suggest we do a `(!this::isDataFileDueToPendingCompaction)`

Reply:
Done